src/audio/kpb.c

```diff
@@ -641,13 +643,15 @@ static int kpb_copy(struct comp_dev *dev)
 		/* In normal RUN state we simply copy to our sink. */
 		sink = kpb->sel_sink;
 
-		buffer_lock(sink, &flags);
+		if (sink)
+			buffer_lock(sink, &flags);
 
 		/* Validate sink */
 		if (!sink || !sink->stream.w_ptr) {
 			comp_err(dev, "kpb_copy(): invalid selector sink pointers.");
 			ret = -EINVAL;
-			buffer_unlock(sink, flags);
+			if (sink)
```

Comment:
Isn't this checked above ? We should try and minimise this and fail if it's NULL (and we expect non NULL)